### PR TITLE
Allow paths for top namespace

### PR DESCRIPTION
**Pull request by ryaminal**
## Description
Adding the forward slash `/` to the filter for top-namespace when normalizing

## Motivation and Context

We would like to create a top level with something like `foo/bar` so that the module can be imported as `import foo.bar.stuff`.

## How Has This Been Tested?

Tested this locally on my mac. Should work on most *nix systems, but I'm not certain how it will impact Windows.
Made sure to have files that actually import with the top-namespace.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] I have read the [code of conduct](https://github.com/davidvujic/poetry-multiproject-plugin/blob/master/CODE-OF-CONDUCT.md).
- [X] I have read the [contributing guide](https://github.com/davidvujic/poetry-multiproject-plugin/blob/master/CONTRIBUTING.md).
- [X] I have updated the documentation accordingly (if applicable).

### CODE REVIEW

**Comment by ryaminal** [comment]
I was really struggling to figure out how to better test this so mistakes like I made in #32 don't happen again.
A simple unit test isn't sufficient, it seems. I wonder how the Poetry project and other Poetry plugins do unit/integration tests? We could do essentially automate what I started to do to test:
 1. create poetry project
 2. create a few files
 3. create another project that will consume project 1
 4. build project 1
 5. install project 1 in to project 2
 6. run a pytest in there
 
 That is so very cumbersome to setup, but not the end of the world. Curious if there is a better way for another PR.

**Comment by DavidVujic** [comment]
Thank you @ryaminal, I'll have a look!

File: poetry_multiproject_plugin/components/parsing/rewrite.py [hunk 1]
```diff
@@ -35,6 +36,7 @@ def mutate_import_from(
 
 
 def mutate_imports(node: ast.AST, namespaces: List[str], top_ns: str) -> bool:
+    ast.FormattedValue
```
Comment (DavidVujic) [inline poetry_multiproject_plugin/components/parsing/rewrite.py:39]: I'm curious of what this do?

File: README.md [hunk 2]
```diff
@@ -59,33 +59,46 @@ poetry build-project --with-top-namespace my_namespace
...
 my_namespace/
     /my_package
        __init__.py
        my_module.py
 ```
 
+###### Namespace with path(`--with-top-namespace=my_namespace/subdir`)
+```shell
+my_namespace/
+    /my_package
```
Comment (DavidVujic) [inline README.md:80]: I think this should be:
/subdir
    /my_package

right?

File: poetry_multiproject_plugin/components/project/prepare.py [hunk 3]
```diff
@@ -34,4 +34,4 @@ def copy_project(project_file: Path, destination: Path) -> Path:
 
 
 def normalize_top_namespace(namespace: Union[str, None]) -> Union[str, None]:
-    return re.sub("[^a-zA-Z_]", "", namespace) if namespace else None
+    return re.sub("[^a-zA-Z_/]", "", namespace.strip("/")) if namespace else None
```
Comment (DavidVujic) [inline poetry_multiproject_plugin/components/project/prepare.py:37]: 👏

File: poetry_multiproject_plugin/components/parsing/rewrite.py (hunk 1 above)
Comment (ryaminal) [reply poetry_multiproject_plugin/components/parsing/rewrite.py:39]: that is just a hanging piece of junk from when i was tinkering around trying to find a better solution to converting a path to a dotted module name. was hoping there was something in ast or various other places but i couldn't find one.
so, i'll remove this line as it's very much not needed.

File: poetry_multiproject_plugin/components/project/prepare.py (hunk 3 above)
Comment (ryaminal) [reply poetry_multiproject_plugin/components/project/prepare.py:37]: i went back and forth on removing the `/` from before/after or just throwing an error and having the user change it. not certain which one is better but i opted for this as it's "easier". :)

File: README.md (hunk 2 above)
Comment (ryaminal) [reply README.md]: grrr, yes. good find.

**Comment by ryaminal** [comment]
We started using a project published with these changes and it appears to be working. Apologies again for the wasted time with my mistakes.

**Comment by DavidVujic** [comment]
Great, thank you @ryaminal! ⭐ I will have a look this weekend.

**Review by DavidVujic — APPROVED** [review]
👏 👏 👏

**Comment by DavidVujic** [comment]
v1.30 is out! Thank you for contributing to this project!

**Pull request merged**